src/components/Notification/__tests__/Notification.test.js

```diff
@@ -0,0 +1,359 @@
+  });
+
+  it('uses timeouts to time navigation actions for VARIANT_SUCCESS', () => {
+    expect(getSetTimeoutCalls()).toHaveLength(0);
+
+    render(
+      withAppContext(<Notification title="Foo bar" variant={VARIANT_SUCCESS} />)
+    );
+
+    act(() => {
+      jest.runAllTimers();
+    });
+
+    expect(getSetTimeoutCalls()).toHaveLength(2);
+  });
+
+  it('does not use timeouts to time navigation actions for TYPE_GLOBAL', () => {
+    expect(getSetTimeoutCalls()).toHaveLength(0);
+
+    render(
+      withAppContext(<Notification title="Foo bar" type={TYPE_GLOBAL} />)
+    );
+
+    act(() => {
+      jest.runAllTimers();
+    });
+
+    expect(getSetTimeoutCalls()).toHaveLength(0);
+  });
+
+  it('hides the component after a specific amount of time and executes callback function', () => {
+    expect(getSetTimeoutCalls()).toHaveLength(0);
+    expect(getClearTimeoutCalls()).toHaveLength(0);
+
+    const onClose = jest.fn();
+
+    const { container } = render(
+      withAppContext(<Notification title="Foo bar" onClose={onClose} />)
+    );
+
+    expect(getSetTimeoutCalls()).toHaveLength(2);
+    expect(getClearTimeoutCalls()).toHaveLength(0);
+
+    expect(onClose).not.toHaveBeenCalled();
+
+    expect(container.firstChild.classList.contains('slideup')).toEqual(false);
+
+    act(() => {
+      jest.advanceTimersByTime(ONCLOSE_TIMEOUT);
+    });
+
+    expect(onClose).not.toHaveBeenCalled();
+
+    act(() => {
+      jest.advanceTimersByTime(SLIDEUP_TIMEOUT);
+    });
+
+    expect(onClose).toHaveBeenCalled();
+
+    expect(getSetTimeoutCalls()).toHaveLength(2);
+    expect(getClearTimeoutCalls()).toHaveLength(2);
+
+    expect(container.firstChild.classList.contains('slideup')).toEqual(true);
+  });
+
+  it('clears timeouts on unmount', () => {
+    expect(getSetTimeoutCalls()).toHaveLength(0);
+
+    const { unmount } = render(
+      withAppContext(<Notification title="Foo bar" />)
+    );
+
+    act(() => {
+      jest.runAllTimers();
+    });
+
+    act(() => {
+      unmount();
+    });
+
+    expect(getClearTimeoutCalls()).toHaveLength(4);
+  });
+
+  it('clears the timer when the component receives a mouse enter event', () => {
+    const onClose = jest.fn();
+
+    expect(getSetTimeoutCalls()).toHaveLength(0);
+
+    const { getByTestId } = render(
+      withAppContext(<Notification title="Foo bar" onClose={onClose} />)
+    );
+
+    expect(onClose).not.toHaveBeenCalled();
+
+    act(() => {
+      jest.advanceTimersByTime(ONCLOSE_TIMEOUT / 2);
+    });
+
+    expect(onClose).not.toHaveBeenCalled();
+
+    expect(getSetTimeoutCalls()).toHaveLength(2);
+    expect(getClearTimeoutCalls()).toHaveLength(0);
+
+    act(() => {
+      fireEvent.mouseOver(getByTestId('notification'));
+    });
+
+    act(() => {
+      jest.runAllTimers();
+    });
+
+    expect(getSetTimeoutCalls()).toHaveLength(2);
+    expect(getClearTimeoutCalls()).toHaveLength(4); // clearTimeout calls on update and unmount
```

Comment:
Could we split this in 2 expect lines. 1 here and one between the `mouseOver()` and `runAllTimers()`

Reply:
I'm going to get rid of the timer checks and going to replace them with a check on the `onClose` function, because that is what I actually should be checking for.